config/routes.rb

```diff
@@ -1,3 +1,31 @@
 Rails.application.routes.draw do
+  # Routes for signing up
+
+  match("/sign_up", { :controller => "sessions", :action => "registration_form", :via => "get"})
```

Comment:
It seems to me that if one were to generate another account, e.g. `leader`, many of these routes would conflict; in particular, `/sign_up`, `/sign_in`, `/verify_credentials`, `/sign_out`, `/edit`, and `/cancel_account`.

Perhaps we should nest these under `/[PLURAL MODEL NAME]/`?

Also, I know I said we should have a separate `sessions_controller`, but I think we also should have a `members_controller` for CRUDing in that table. `sessions_controller` would be responsible for CRUDing in the cookie jar.

Together, and making it resemble the actions from a standard `draft:resource`, that would make the routes look something like:

```ruby
  match("/members/sign_up", { :controller => "members", :action => "new_registration_form", :via => "get"})

  match("/members/insert_record", { :controller => "members", :action => "create", :via => "post" })
          
  # Routes for signing in
  match("/members/sign_in", { :controller => "member_sessions", :action => "session_form", :via => "get"})
  
  match("/members/verify_credentials", { :controller => "member_sessions", :action => "add_cookie", :via => "post"})
  
  # Route for signing out
  
  match("/members/sign_out", { :controller => "member_sessions", :action => "remove_cookies", :via => "get"})

  # Route for editing account
  
  match("/members/edit", { :controller => "members", :action => "edit_registration_form", :via => "get"})
  
  match("/members/update_record", { :controller => "members", :action => "update", :via => "post"})
  
  # Route for removing an account
  
  match("/members/cancel_account", { :controller => "members", :action => "destroy", :via => "get"})
```

Reply:
Nesting the routes as in the above could lead to problems when a student adds a show page, e.g. `/members/:id`. They could mess up the order.

Perhaps we should go with `/member_sign_up`, `/member_sign_in`, etc?

Reply:
Yes, that makes sense.